internal/pkg/deploy/cloudformation/stack/scheduled_job.go

```diff
@@ -0,0 +1,315 @@
+
+		parser: parser,
+	}, nil
+}
+
+// Template returns the CloudFormation template for the scheduled job.
+func (j *ScheduledJob) Template() (string, error) {
+	outputs, err := j.addonsOutputs()
+	if err != nil {
+		return "", err
+	}
+
+	sidecars, err := j.manifest.Sidecar.Options()
+	if err != nil {
+		return "", fmt.Errorf("convert the sidecar configuration for job %s: %w", j.name, err)
+	}
+
+	schedule, err := j.awsSchedule()
+	if err != nil {
+		return "", fmt.Errorf("convert schedule for job %s: %w", j.name, err)
+	}
+
+	stateMachine, err := j.stateMachineOpts()
+	if err != nil {
+		return "", fmt.Errorf("convert retry/timeout config for job %s: %w", j.name, err)
+	}
+
+	content, err := j.parser.ParseScheduledJob(template.WorkloadOpts{
+		Variables:          j.manifest.Variables,
+		Secrets:            j.manifest.Secrets,
+		NestedStack:        outputs,
+		Sidecars:           sidecars,
+		ScheduleExpression: schedule,
+		StateMachine:       stateMachine,
+		LogConfig:          j.manifest.LogConfigOpts(),
+	})
+	if err != nil {
+		return "", fmt.Errorf("parse scheduled job template: %w", err)
+	}
+	return content.String(), nil
+}
+
+// awsSchedule converts the Schedule string to the format required by Cloudwatch Events
+// https://docs.aws.amazon.com/lambda/latest/dg/services-cloudwatchevents-expressions.html
+// Cron expressions must have an sixth "year" field, and must contain at least one ? (either-or)
+// in either day-of-month or day-of-week.
+// Day-of-week expressions are zero-indexed in Golang but one-indexed in AWS.
+// @every cron definition strings are converted to rates.
+// All others become cron expressions.
+func (j *ScheduledJob) awsSchedule() (string, error) {
+	if j.manifest.Schedule == "" {
+		return "", fmt.Errorf("missing required field \"schedule\" in manifest for job %s", j.name)
+	}
+
+	// Try parsing the string as a cron expression to validate it.
+	if _, err := cron.ParseStandard(j.manifest.Schedule); err != nil {
+		return "", fmt.Errorf("schedule %s for job %s is not a valid cron expression, rate, or preset schedule", j.manifest.Schedule, j.name)
+	}
+	var scheduleExpression string
+	var err error
+	switch {
+	case strings.HasPrefix(j.manifest.Schedule, every):
+		scheduleExpression, err = toRate(j.manifest.Schedule[len(every):])
+		if err != nil {
+			return "", fmt.Errorf("parse fixed interval: %w", err)
+		}
+	case strings.HasPrefix(j.manifest.Schedule, "@"):
+		scheduleExpression, err = toFixedSchedule(j.manifest.Schedule)
+		if err != nil {
+			return "", fmt.Errorf("parse preset schedule: %w", err)
+		}
+	default:
+		scheduleExpression, err = toAWSCron(j.manifest.Schedule)
+		if err != nil {
+			return "", fmt.Errorf("parse cron schedule: %w", err)
+		}
+	}
+	return scheduleExpression, nil
+}
+
+// toRate converts a cron "@every" directive to a rate expression defined in minutes.
+// example input: @every 1h30m
+//        output: rate(90 minutes)
+func toRate(durationString string) (string, error) {
```

Comment:
nit: reading `durationString string` (string twice) feels redundant. I think in this situation we can actually just do 
```suggestion
func toRate(duration string) (string, error) {
```